homeassistant/components/aurora_abb_powerone/__init__.py

```diff
@@ -69,36 +71,43 @@ def _update_data(self) -> dict[str, float]:
         """
         data: dict[str, float] = {}
         self.available_prev = self.available
-        try:
-            self.client.connect()
-
-            # read ADC channel 3 (grid power output)
-            power_watts = self.client.measure(3, True)
-            temperature_c = self.client.measure(21)
-            energy_wh = self.client.cumulated_energy(5)
-        except AuroraTimeoutError:
-            self.available = False
-            _LOGGER.debug("No response from inverter (could be dark)")
-        except AuroraError as error:
-            self.available = False
-            raise error
-        else:
-            data["instantaneouspower"] = round(power_watts, 1)
-            data["temp"] = round(temperature_c, 1)
-            data["totalenergy"] = round(energy_wh / 1000, 2)
-            self.available = True
-
-        finally:
-            if self.available != self.available_prev:
-                if self.available:
-                    _LOGGER.info("Communication with %s back online", self.name)
-                else:
-                    _LOGGER.warning(
-                        "Communication with %s lost",
-                        self.name,
-                    )
-            if self.client.serline.isOpen():
-                self.client.close()
+        retries: int = 3
+        while retries > 0:
+            try:
+                self.client.connect()
+
+                # read ADC channel 3 (grid power output)
+                power_watts = self.client.measure(3, True)
+                temperature_c = self.client.measure(21)
+                energy_wh = self.client.cumulated_energy(5)
+            except AuroraTimeoutError:
+                self.available = False
+                _LOGGER.debug("No response from inverter (could be dark)")
+                retries = 0
+            except (SerialException, AuroraError) as error:
+                self.available = False
+                retries -= 1
+                _LOGGER.warning(
```

Comment:
This should not log on a warning level.

The same for the ones below actually.

We should only log 1 time, of which the DataUpdateCoordinator takes care. Just raise an `UpdateFailed` (instead of logging manually).

Reply:
Thanks for the review!

Ok, I understand that no explicit logging is needed here. 
If I raise an error, it won't retry immediately - I was hoping to prevent gaps in data caused by communication glitches which seem to happen with this device.  I don't see a way of raising an exception and also retrying.

If the solution is to wait for the next update interval then the retry logic is probably unnecessary.

The issue of raising an error vs warning came up on the previous PR. https://github.com/home-assistant/core/pull/72363#discussion_r1305557202.  I'm trying to not generate errors during normal operation (sunset).